### PR TITLE
gamemode: 1.6.1 -> 1.7

### PR DESCRIPTION
**Pull request by azahi**
###### Description of changes

https://github.com/FeralInteractive/gamemode/releases/tag/1.7

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/games/gamemode/default.nix [hunk 1]
```diff
@@ -14,34 +14,22 @@
...
-    # See https://github.com/FeralInteractive/gamemode/pull/324
-    (fetchpatch {
-      url = "https://github.com/FeralInteractive/gamemode/commit/b29aa903ce5acc9141cfd3960c98ccb047eca872.patch";
-      sha256 = "LwBzBJQ7dfm2mFVSOSPjJP+skgV5N6h77i66L1Sq+ZM=";
-    })
-
     # Add @libraryPath@ template variable to fix loading the PRELOAD library
     ./preload-nix-workaround.patch
+    # Do not install systemd sysusers configuration
+    ./no-install-systemd-sysusers.patch
```
Comment (kira-bruneau) [inline pkgs/tools/games/gamemode/default.nix:32]: Hmm, yeah we definitely don't want this in the Nix package. The NixOS module uses the packaged polkit policies to let you run `cpugovctl` & `gpuclockctl` without root. Although, maybe we should consider refactoring that to require your user to be a part of the gamemode group, so not everyone can run them? (not for this PR though)

Either way, I think a meson option should be added upstream to disable it so we could get rid of the patch in a future update.

**Review by kira-bruneau — COMMENTED** [review]
:heart: Thanks @azahi!

Just one thing I noticed is that the new `gamemodelist` command is a shell script, and we should probably encode the dependencies it relies on:

```diff
diff --git a/pkgs/tools/games/gamemode/default.nix b/pkgs/tools/games/gamemode/default.nix
index 51c066260b52..d15ae78b17a8 100644
--- a/pkgs/tools/games/gamemode/default.nix
+++ b/pkgs/tools/games/gamemode/default.nix
@@ -3,6 +3,7 @@
 , fetchFromGitHub
 , fetchpatch
 , libgamemode32
+, makeWrapper
 , meson
 , ninja
 , pkg-config
@@ -10,6 +11,9 @@
 , inih
 , systemd
 , appstream
+, findutils
+, gawk
+, procps
 }:
 
 stdenv.mkDerivation rec {
@@ -43,6 +47,7 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [
+    makeWrapper
     meson
     ninja
     pkg-config
@@ -73,13 +78,20 @@ stdenv.mkDerivation rec {
     moveToOutput lib/*.a "$static"
   '';
 
-  # Add $lib/lib to gamemoded & gamemode-simulate-game's rpath since
-  # they use dlopen to load libgamemode. Can't use makeWrapper since
-  # it would break the security wrapper in the NixOS module.
   postFixup = ''
+    # Add $lib/lib to gamemoded & gamemode-simulate-game's rpath since
+    # they use dlopen to load libgamemode. Can't use makeWrapper since
+    # it would break the security wrapper in the NixOS module.
     for bin in "$out/bin/gamemoded" "$out/bin/gamemode-simulate-game"; do
       patchelf --set-rpath "$lib/lib:$(patchelf --print-rpath "$bin")" "$bin"
     done
+
+    wrapProgram "$out/bin/gamemodelist" \
+      --prefix PATH : ${lib.makeBinPath [
+        findutils
+        gawk
+        procps
+      ]}
   '';
 
   meta = with lib; {
```

File: pkgs/tools/games/gamemode/default.nix (hunk 1 above)
Comment (azahi) [reply pkgs/tools/games/gamemode/default.nix:36]: I'm not exactly sure how to approach the process of adding a group, giving it appropriate permissions and then somehow prompt the user to add themselves to it, so I'll leave this part as-is.

**Comment by kira-bruneau** [comment]
Thanks @azahi!

**Pull request merged**